lib/workers/branch/index.ts

```diff
@@ -142,8 +142,9 @@ export async function processBranch(
         return 'needs-approval';
       }
     }
+    const wouldCommit = branchExists ? branchPr?.isStale : true;
```

Comment:
We don't always commit if a branch is stale. Or is there an outer if I'm missing on mobile?

Reply:
This is my naїve criteria, which should be more specific. It's hard for me to infer it from the code, since the function is pretty big. Also I'm concerned with duplication of `if-else` logic if the desired criteria is complex. Probably you can say clearly which condition leads to commit.

Another option is to (1) remove the limit condition from it, (2) compute the result of `processBranch()` with dry-run option, then (3) check limits for specific results like `pr-edited`, and finally (4) call `processBranch()` as usual. It can work if dry-run `processBranch()` result can say if it would commit or not.

Reply:
Needs to be correct or we shouldn't merge this. Otherwise can cause other problems.

Reply:
Agree, will do more research on this.